### ISSUE TITLE
Community creation

### ISSUE THREAD

**Issue by jefgodesky** [jefgodesky/thefifthworld-react#36]
### Step 1: Where you at?

Base level here is just a textbox asking for your latitude and longitude. It's pretty bad, but what else can we do if JS fails? If JS works, we'll show a map, and request geolocation. If granted, we'll default to where the member is, and ask, "Is this where your community is?" They'll have the option to change it, but if we're doing this right, they shouldn't most of the time. If they have JS but deny geolocation, then you get a map and you need to choose where this community lives, roughly.

We use that to determine if this is a coastal community (draw a 20-mile radius circle centered on the point we were given; if that circle intersects our coastline polygons, it's coastal) or if it has access to rivers (if that same 20-mile radius circle intersects any rivers).

### Step 2: What do you emphasize?

Pretty much everybody in the Fifth World generalizes, but what foods do you emphasize? You have a reputation for being really good at ______. Pick ... ah'dunno, 4? Something like that.

The end of seasonality means that there isn't much difference from season to season, and a heterogeneous jungle planet from pole to pole means it's pretty much the same options everywhere you go. The diversity comes from the fact that you can't emphasize everything, so you'll end up in complex political-economic relationships with the people who emphasize the things that you don't — and perhaps in rivalries with those who emphasize the same things you do. So we get a tangled web of intrigue this way.

This is where figuring out if our communities has access to the coast and/or rivers comes back into play, as this will change the list of options available.

### Step 3: Discover some places.

Based on the things that you choose in the previous step, we can put your community somewhere on a spectrum from horticulture to foraging. This lets us estimate population density and territory size.

It also lets us pick a few key cards from the deck that are particularly important to you. For example, if you grow a lot of plants, we'll want to name the Garden right from the start, whereas if you hunt a lot of animals, we'll want to name the Hunting Grounds instead.

If JS is working, you'll get a map, showing your territory (by relying on that estimate of territory size we were able to make, based on what you emphasize). Otherwise, you'll need to enter latitude and longitude again, which, again, will suck. We can make sure that the places you add aren't too far away, since we have that estimated territory size.

### Step 4: Approve community.

OK, so now there's some pretty involved stuff that happens behind the scenes. We'll have to work out how big an initial pool we'll need to get to the population size we're expecting in the end, but whatever that number is, we're going to create that many random strings — strings like this:

```
-+-+0000-0-0+-0--0-+----0+-+-+0++0----000-++0-++0+--0--++-0-0+00++----0-RDDRDDX
```

If you've ever worked on a [genetic algorithm](https://en.wikipedia.org/wiki/Genetic_algorithm), you might know where I'm going with this.

* *Characters 1-4:* These are determining _openness to experience_, one of the Big Five personality traits. They're read like the results of Fudge dice: a "-" means -1, a "0" means +0, and "+" means +1. The results of these four characters are added together to get the character's total openness to experience rating, which is close to a normal distribution curve with a mean of 0. The maximum is 4, and the minimum is -4.
* *Characters 5-8:* Conscientiousness, another of the Big Five Personality traits, also recorded like Fudge dice.
* *Characters 9-12:* Extraversion, another of the Big Five Personality traits, also recorded like Fudge dice.
* *Characters 13-16:* Agreeableness, another of the Big Five Personality traits, also recorded like Fudge dice.
* *Characters 17-20:* Neuroticism, another of the Big Five Personality traits, also recorded like Fudge dice.
* *Characters 21-24:* Intelligence, also recorded like Fudge dice.
* *Characters 25-28:* Neurodivergence, also recorded like Fudge dice.
* *Characters 29-32:* Sexual orientation, also recorded like Fudge dice. A -4 here would be a 0 on the Kinsey scale (exclusively heterosexual), while a 4 here would be a 6 on the Kinsey scale (exclusively homosexual). Yes, with the normal distribution here, this is taking the side that pretty much all of us are mostly bisexual, and so are more or less willing to accept whatever the social norm is.
* *Characters 33-36:* Cluster A personality disorders. If neuroticism is high and these are also very high, then we have an individual with a Cluster A personality disorder. We want to make sure we calibrate those thresholds so that the mathematical incidence of cluster A personality disorders works out to be about right.
* *Characters 37-40:* Cluster B personality disorders, treated much like cluster A personality disorders, earlier.
* *Characters 41-44:* Cluster C personality disorders, treated much like cluster A personality disorders, earlier.
* *Characters 45-48:* Height
* *Characters 49-52:* Body type, from ectomorph (-4) to endomorph (4).
* *Characters 53-56:* Eye color. We use a random chart used in Fate games to assign different colors to different outcomes. We hope this helps mirror the unpredictability of recessive traits and multiple genes impacting traits like this.
* *Characters 57-60:* Hair color, treated much like eye color.
* *Characters 61-64:* Skin color, treated much like eye color.
* *Characters 65-68:* Relative testosterone levels.
* *Characters 69-72:* Relative estrogen levels.
* *Characters 73-74:* Blindness. This is either an "R" for recessive or a "D" for dominant. If you have two R's, you're born blind, but we'll be sure that our starting population is seeded with R's proportional to the number of people born blind today.
* *Characters 75-76:* Deafness, treated like blindness, earlier.
* *Characters 77-78:* Muteness, treated like blindness, earlier.
* *Character 78:* X/Y chromosome. One X can be assumed, so X here means XX (chromosomally female), and a Y here means XY (chromosomally male).

So our first generation is a bunch of randomly-generated strings, with some input to reflect incidence of modern-day people. We apply some defined generational pressures (e.g., 10% of psychopaths die before they can reproduce). Then, from among the survivors, we pair up men and women and generate some random number of children. Each child builds a string by randomly taking each character from either its mother or its father. That next generation faces other generational pressures, but it goes through the same thing again. And again and again until we've reached the present of the Fifth World, 13 generations later.

__Why in Xandru's Hells would we do this?!?__ Well, a couple of reasons.

# *Getting representation right is really hard.* Way too hard to expect human beings to get right all on their own. It's very Scylla and Charybdis. You can try too hard to make sure that everyone is represented and end up with something that makes the exceptional seem commonplace. Or you can try too hard to make everything typical, and wind up with something that makes it seem like exceptional people don't exist at all. Exceptional people exist, but they're exceptional. I've yet to meet the person whose intuitive sense of this was right on. Mine certainly isn't. I've made both errors at various times in my own games. When I started writing algorithms like this to help me, things got really, really good. So I'm building it into the Fifth World.
# *This can help us better model what the Fifth World will actually look like.* This sort of modeling might point out unexpected things. We may well find that we need to tweak our algorithm, of course. Those generational pressures really stick out to me as something that we'll really need to spend some time on. But it has the potential to really teach us something.
# *I expect this will do wonders for our blank page problem.* I mentioned before that I started making generators like this to ensure that I was representing gender, sexual, ethnic, disabled, and other identities fairly in my games, but as a side effect, __we got way better games__. My generators consistently delivered fun characters that I honestly never would have come up with by myself. The human brain looks for patterns and storylines. I think of it like smoothing things over, figuring out how things make sense. Left to itself, it produces something nice and smooth, something that makes sense, something consistent and thematic and reasonable. Random generators create static. They create weird juxtapositions and contradictions and things that don't seem to make sense. Our human brains deal with input like this from the real world all the time, and the stories we tell apply that "smoothing" to the chaos of reality to produce something beautiful. But that means it's the application of the human brain's smoothing effect to something jagged and chaotic that gets us there. Generators like this one create that jagged, chaotic reality for our minds to work on. They give us contradictions to figure out, opposites to unite, and ultimately stories to tell. When you look at the characters produced by a generator, it usually doesn't take long for the stories to start leaping out at you.

So, we go through this whole process, and we end up with our community. We'll need to figure out some way to display this for approval. Let the member tweak things as they like, especially the names of characters and how they're described.

When you approve, the community, its places, and its members all get added to the site as their own pages, and you land on your shiny new community page.

## How I'm going to do this

Well, there's some theoretical work to be done here first, isn't there? You'll notice a few questions I just left dangling, unanswered. I'm going to have to answer those. That work's going to happen in the comments below.

And it's not like any of this is particularly useful until it's all done, so we're going to be working on this in the `features/34-community-creation` branch.

**Comment by jefgodesky** [comment]
## Options to emphasize

### Cutting rivers

Here's one of the most dire things I've had to face up to: the rivers in the Fifth World might be almost entirely dead. Some carp and bass might be OK, but I'm having a hard time finding any other examples. Freshwater species tend to be pretty niche creatures, adapted to particular temperate and flow conditions. The oceans are all connected, so as conditions change you can move, but that's not always true with freshwater environments. If the climate changes, quite often, there's just nowhere to go. So we're not going to try to figure out if you have any rivers nearby, because the sad truth is, it's unlikely to change anything for you.

### Starting point

My starting point here is a quick run-down of foods preferred. by the Aché in Paraguay and the Mbuti in the Congo. They're both hunter-gatherers living in tropical environments, but they're on different continents and have very different approaches. They're also heavily studied, with lots of quantitative data to look at. So I've been figuring, any community creation that could be used to create an Aché community _or_ a Mbuti community is probably a good starting point.

#### Mbuti

* Crabs
* Shellfish
* Ants
* Larvae
* Snails
* Pigs
* Antelopes
* Monkeys
* Wild yams
* Berries
* Fruits
* Leaves
* Kola nuts

#### Aché

* Armadillo
* Monkeys
* Deer
* Collared peccary
* Tegu lizard
* Tapir
* White-lipped peccary

### Feral species

Then there's domesticated animals that have gone feral, giving us that post-apocalyptic vibe.

* Cows
* Pigs
* Chickens
* Horses
* Sheep

### Base list

If each community can pick 4, and we want to make sure we have a good amount of variety in here, we'll need at least 20 for the base list. I got to 18 using the sources above, and it seemed a little animal-heavy anyway, so I added cassava, the three sisters, and all the plants that Giuli mentioned as lending their names to a generation in [chapter 1](https://thefifthworld.com/wormwood/01) of _[Children of Wormwood](https://thefifthworld.com/wormwood)_: tomato, chili pepper, onion, garlic, banana, sweet potato, lemongrass, ginger, coconut, coffee, cocoa, and vanilla. And while I was at it, I really love potatoes and I love what James C. Scott has to say about the impact of tubers like potatoes in _The Art of Not Being Governed_, so potatoes make the cut, too. That gives us a base list of 32. I thought we might want to make it a nice round 25, so I cut horses, armadillos (even though the Aché apparently rely really heavily on them, that might be a niche thing), ants and larvae (everyone eats insects, but I'm not sure anyone would emphasize it), and leaves (a little too broad, there).

* Banana
* Berries
* Cassava
* Chickens
* Chili peppers
* Cocoa
* Coconut
* Coffee
* Cows
* Deer
* Fruits
* Garlic
* Ginger
* Lemongrass
* Lizards
* Monkeys
* Nuts
* Onion
* Pigs
* Potato
* Sheep
* Sweet potato
* Three sisters
* Tomato
* Vanilla

### Coastal options

If you're on the coast, there's five more options, which could bring your total options up to 30. Mbuti eat crabs and shellfish, but ocean acidification and climate change don't bode well for them. In fact, [global warming will hit the oceans even harder than the land](https://insideclimatenews.org/news/24042019/climate-change-fish-local-extinction-marine-species-hit-harder-land-animals-study-nature). But ecology is never simple enough to be summed up with _just_ doom and gloom. One species's loss is always some other species's gain, right? In _[Children of Wormwood](https://thefifthworld.com/wormwood)_ Giuli describes the Nares people not as fisherfolk but as squidders, because [cephalopods appear to be doing well under these conditions](https://www.theguardian.com/environment/2016/may/23/octopuses-squid-cuttlefish-warming-oceans-climate-change). So we'll put squid, cuttlefish, annd octopi on the list. [Jellyfish seem to be benefiting from climate change](https://www.weforum.org/agenda/2019/01/how-an-explosion-of-jellyfish-is-wreaking-havoc/), so we'll add them to the list, too. [How global warming will affect sharks is up in the air](https://oceana.org/blog/climate-change-giant-question-mark-future-sharks), but I'm going to go with the possibility that they make it and are abundant enough to be a regular source of food, because (a) they're some of the most ancient creatures in the seas and have survived plenty of apocalypses before, and (b) they eat squids, octopi, and jellyfish, so if those species are doing well, then sharks won't be hurting for food.

* Squid
* Cuttlefish
* Octopi
* Jellyfish
* Sharks

**Comment by jefgodesky** [comment]
## Places to discover

Everyone’s going to discover these places:

* Axis Mundi (C10)
* Scout’s Nest (H7)
* Monument (S7)

If you picked bananas, cassava, cows, pigs, potatoes, sheep, sweet potatoes, or the three sisters for foods that you emphasize, then you’re a horticultural village (you might hunt feral cows, sheep, or pigs as a hunter-gatherer, but to *emphasize* them you’ll need to keep a semi-domesticated herd). Otherwise, you’re a hunter-gatherer band. This will determine the diamonds place you discover.

&nbsp; | Hunter-gatherer band | Horticultural village
------------: | :------------: | :-------------:
**Average population** | 30 | 150
**Average territory area** | 90 sq. miles | 30 sq. miles
**Average population density** | 0.33/sq. mile | 5/sq. mile 
**Average territory radius** | 5.5 miles | 3 miles	
**Place** | Hunting ground | Garden
**Card** | D5 | D7

These cards should help flesh out some important details about your community.

*  Axis Mundi touches on how you relate to the communities around you, and what lies at the center of your world, suggesting or outright describing your cosmology.
* Scout’s Nest gets us into scouts and the role they play, which tells us a lot about the things that worry you, suggesting sources of tension and concern.
* Monument digs into your community’s history and how you understand and relate to your past.
* Hunting ground gets us into more specifics about what you hunt and how you hunt.
* Garden gets us into your land management and how you relate to the land you tend and the plants you grow.

So with these places, we have your community off to a good start from a gameplay perspective and we’ve filled in some broad strokes about your community’s traditions, history, values, and concerns. Hopefully, this will create a lot of hooks in a way that’s fairly straightforward and really leads you by the hand along the way.

**Comment by jefgodesky** [comment]
## Creating a community

Our top-level cycle will be time, measured in years. The “present” of the Fifth World is always 144,000 days from today, a little more than 394 years from now. We’ll start 150 years before that, which would be 244 years from now. So, for example, I’m writing this in 2019, so if I were to run this right now it would start with 2263 and run to 2413.

The number of ancestors we start working from differs by community type. We’ll start by marking that many years in the first 25 years that we’re cycling through as years to generate a new character. For example, a hunter-gatherer band has four ancestors, so we pick four numbers between 1 and 25 randomly and get 24, 12, 18, and 14. So we mark years 12, 14, 18, and 24. We start cycling. Year 1, nothing happens because we don’t have anyone in our community yet. Year 2, same. Year 3, same, until we get to year 12. At year 12, we generate our first community member, so in year 13 we apply our first random event to that person’s life. In year 14, that person gets another random event, and we add a second community member.

Community | Ancestors
---|:---:
Hunter-gatherer band | 4
Horticultural village | 20

And then we just keep doing that until we reach the “present” of the Fifth World in 2413, and we see what we have (the problem of how we actually _display_ all of this I’m leaving for the next installment).

### Traits

We give each character some traits to begin with.

* **Big Five Personality Traits:** We use these in combination to create a description of the character’s personality. Each one is normally distributed, with a mean of zero and a standard deviation of one. All traits start at zero.
* **Intelligence:** Random value from a normal distribution with a mean of zero and a standard deviation of one. Measuring intelligence in actual people is fraught with difficulties and problems (What kinds of intelligence do you measure? Does your test actually measure those types of intelligence, or something else, like compliance? Is your test really just a bunch of garbage cooked up by eugenicists, like IQ is?). But we’re not really measuring anything here, because these are fictional people. We use the value to give a descriptor, so the value here is creating a normal distribution of the people who are described as smart versus the number of people who are described as not very bright. And we’re probably not going to use it at all unless it’s Ann exceptional value.
* **Neurodivergence:** Random value from a normal distribution. If you’re in the top 1% of that distribution, you’d be considered on the autism spectrum today. We’ll probably use two standard deviations (closer to 2%) as our cut-off to mention something in the description.
* **Sexual orientation:** Random value from a normal distribution, mapped to the Kinsey scale. This is taking the side that pretty much all of us are mostly bisexual, and so are more or less willing to accept whatever the social norm is. In today’s heteronormative society, that means 90% of us are heterosexual.
* **Sex:** 49.15% female, 49.15% female, and 1.7% intersex.
* **Gender:** For people with male or female gender, 99% are cisgender, 1% are third gender.
* **Body type:** This is formed from a string of four bits. Each one is one of [+0-], where “+” means +1, “0” means +0, and “-“ means -1. These are added together to create a value between -4 and +4, normally distributed, with a mean of 0. The lower the value, the more ectomorphic one’s body. The higher the value, the more endomorphic. This is only generated randomly for the first generation; afterwards, it’s generated by taking the mother’s and father’s values, and choosing randomly from one or the other for each bit, providing inheritance (children will tend to resemble their parents) with variation (a completely average father with -+-+ and a completely average mother with +-+- might still end up with an extremely ectomorphic child with ----; it’s just not the most likely outcome).
* **Limbs:** You have right and left arms, and right and left legs. 0.1% of characters will have one of these marked as disabled at creation (dysmelia).
* **Blindness:** 0.016% of characters are marked as blind at creation.
* **Deafness:** 0.2% of characters are marked as deaf at creation.
* **Achondroplasia:** 0.004% of characters will have achondroplasia, making them little people. This is only generated randomly for the first generation. Thereafter, if neither parent has achondroplasia, it remains a 0.004% chance. If one has achondroplasia but not the other, it rises to 50%. If both parents have it, then there’s a 50% chance of achondroplasia and a 25% chance that the child dies.
* **Psychopathy:** 1% of characters are psychopaths. This will also provide a -2 to Agreeableness and Conscientiousness, +2 to Neuroticism, and +1 to Extraversion and Openness to experience. This is set to `null` if the character is not a psychopath, but starts at `1` for those who are. See _Psychopathy_ for special procedures for these characters.

### Random events

First, let’s talk about carrying capacity. This differs by community type.

Community | Carrying capacity
---|:---:
Hunter-gatherer band | 30
Horticultural village | 150

First, the community itself has a random event each year. Some events are more likely if the community’s population exceeds its carrying capacity. In the table below, **%** indicates the probability of this event under normal circumstances, and **%O** indicates the probability of this event when the community exceeds its carrying capacity.

% | %O | Event
:---:|:---:|---
1 | 50 | Lean times
1 | 25 | Sickness
1 | 10 | Conflict

Then, each member of the community gets a random event. In the table below, **%** indicates the probability of this event under normal circumstances, and **%O** indicates the probability of this event when the community is experiencing learn times, sickness, or conflict.

% | %O | Event 
:---:|:---:|---
10 | 2 | +1 to Openness to Experience, to a maximum of 3. This can trigger a _family event_ (see below).
10 | 2 | -1 to Openness to Experience, to a minimum of -3.
10 | 2 | +1 to Conscientiousness, to a maximum of 3.
10 | 2 | -1 to Conscientiousness, to a minimum of -3.
10 | 2 | +1 to Extraversion, to a maximum of 3. This can trigger a _family event_ (see below).
10 | 4 | -1 to Extraversion, to a minimum of -3. This can trigger a _breakup_ (see below).
10 | 2 | +1 to Agreeableness, to a maximum of 3. This can trigger a _family event_ (see below).
10 | 4 | -1 to Agreeableness, to a minimum of -3. This can trigger a _breakup_ (see below).
5 | 4 | +1 to Neuroticism, to a maximum of 3. This can trigger a _breakup_ (see below).
10 | 1 | -1 to Neuroticism, to a minimum of -3.
1 | 35 | Sickness. The character has a 5% chance of dying.
1 | 25 | Injury. See _Injury_ for procedure.

#### Family events

Three of the random events above can trigger _family events_. These only happen when the character is 20 years of age or older.

* First, see if the character might have a child (see _Reproduction_)
* Then, see if she might find a partner (see _Matchmaking_)

##### Reproduction

To be eligible, a character must either be a male with a female partner, or female with a male partner, and the mother must not have had another child in the past 4 years. If this is true, we start with a 90% chance, and cut it in half for each child that the character’s family has. If we pass all of that, they have a child. This counts as the random event for this year for both parents and any others they are in a relationship with (e.g., a man and his wife, or a man and his three wives, or four people of various genders in a group relationship).

We generate a new character for the child. Most traits are assigned randomly, except for body type and achondroplasia, which are influenced by the parents.

##### Matchmaking

If a family event is triggered but the character can’t reproduce, she might find a new partner.

If the character has no partner and the community’s population exceeds its carrying capacity, the character leaves the community, joining her partner in some other community.

If the character has no partner and the community’s population does not exceed its carrying capacity, the character has a 50% chance of leaving the community to join her partner in some other community.

Otherwise, we start with a 90% chance, and cut it in half for each partner that the character already has. If she passes, we’ll generate 10 candidates, with genders proportional to the character’s sexual orientation. We then calculate personality distance by taking the difference between their scores on each of the Big Five personality traits and adding those together. We then give each one an attraction score equal to that personality difference minus a random number between 0 and 1 (reflecting that, hey, the heart wants what the heart wants, right?). The candidate with the lowest number wins.

### Breakup

Three of the random events above can trigger _breakups_. When these arise for a character who has one or more partners, check on each one. Calculate the personality distance, just as we did when we formed the relationship, by taking the difference between each individual’s score on each of the Big Five personality traits and adding them together. Multiply this by 10 to get the probability that a separation occurs.

### Skills

This is probably no more true than the “10,000 hours” canard, but let’s say it does take [7 years to master a skill](http://www.smbc-comics.com/?id=2722). You spend your first seven years mastering the skills of just living in the world: walking, talking, and all of that. You spend your next 7 years mastering the more specific skills of engaging with your society. After that, it’s up to you.

Here, I’m going to leave the details a little fuzzy to fill in later, but we’ll have a list for young adults, a list for middle-aged people, and a list for elders. We take a random skill from the appropriate list, based on age, and we set a time equal to the current year plus learning time. Learning time will be equal to seven minus the character’s intelligence score (so intelligent characters pick up skills more quickly).

If you are learning a skill, we check the current year against the year when you’re supposed to finish. If the current year is greater than or equal to that, we add it to the list of skills that you’ve mastered. You now count as not learning anything, so next year you’ll start on something new.

### Injury

% | Injury
:---:|---
2 | Loss of vision. The character is blind.
2 | Loss of left eye. If the right eye was already lost, the character is blind.
2 | Loss of right eye. If the left eye was already lost, the character is blind.
2 | Loss of hearing. The character is deaf.
2 | Loss of left ear. If the right ear was already lost, the character is deaf.
2 | Loss of right ear. If the left ear was already lost, the character is deaf.
4 | Loss of left arm.
4 | Loss of right arm.
4 | Loss of left leg.
4 | Loss of right leg.
3 | The character dies.
9 | The injury becomes infected, and the character dies.
60 | The character takes a scar.

**Comment by jefgodesky** [comment]
OK, so at this point I still need details on skills and to figure out how the hell I'm going to display all of that information I generated with that procedure outlined above.

**Comment by jefgodesky** [comment]
Made some edits to that last big comment to call out some specific calculations that get re-used and plug in more equations instead of thresholds, so that stuff becomes more likely as things progress, rather than just reaching some point where things just happen.

You might be looking at this and thinking there's a vicious cycle with discord, where once you get into it things just get worse and worse. That might be true, I'll have to test it to see. My thinking is that you get on the discord train when your population exceeds carrying capacity, and the discord leads you towards people leaving or dying. Which will bring you back under carrying capacity. Which makes it really likely (97%, to be precise) that you'll experience peace and prosperity. Which brings your discord back down. Which increases your birth rate. It's supposed to be, effectively, a Lotka-Volterra cycle, where the predator is discord.

**Comment by jefgodesky** [comment]
## Displaying results

We want to show the community we came up with and give the user a chance to either accept it, or run it over again. If accepted, we’ll be creating a page for the community and all its members, so we want to make sure that it’s good before we go making all those pages.

### Chronicle

Since we went year by year, we can provide a yearly chronicle of the community’s history.

* Peace is the default, so it doesn’t need to be mentioned. “Time of conflict,” “time of sickness,” and “lean times” do, though.
* Years of peace with exceptional discord will warrant mention.
* We’ll mention community members who were born and those who died.
* Injuries will warrant mention.
* Affairs will warrant mention.
* If a psychopath is caught, we’ll mention that, as well as whether he was killed or exiled.

### Relationship map

We’ll need to figure out a good solution to generate a graphic, but we’ll generate a relationship map, showing parents and children, and who’s in a relationship with who. This one could take a while to get just right, but it’s clearly the best way to present what’s going on. We’ll only show the “current state” of the community.

### Character listing

Then we can just list the characters.

* **Age:** List the character’s age, and her date of birth.
* **Personality:** We’ll need to come up with a short description for the personality of a person who is high, medium, or low on each of the Big Five traits. That gives us [244 combinations to fill in](https://docs.google.com/spreadsheets/d/1-goO2zjq2dNvEO2q5ShrlLoFgVEaCmz6kz-jCXUplNY/edit?usp=sharing). I’ve already asked Giuli to take on this part of it. 
* **Intelligence:** We’ll come up with descriptions for high or low intelligence.
* **Neurodivergence:** We’ll need to come up with a good description for someone who’s neurodivergent.
* **Gender & sexual orientation:** Most people will be “Pansexual man” or “Pansexual woman,” but there’s possibilities for heterosexual or homosexual people, or third gender people.
* **Physical description:** Uses body type, limbs, achondroplasia, and injuries.
* **Skills:** List the character’s skills.
* **Relationships:** List the character’s relationships.
* **Children:** List the character’s children.
* **Psychopathy:** Mention if this person is a psychopath, and if so, how many acts he’s committed.

**Comment by jefgodesky** [comment]
## Skills

Up to age 7, you’re learning the basics of life — walking, talking, and so on. From age 8-14, you’re mastering the specifics of cultural life in your community. Everyone in the community age 15 or older is assumed to have the basic level of skills needed for life in that community. So, for example, every member of a hunter-gatherer band can hunt and track as well as the best modern hunters, they know and can mimic animal and bird calls fluently enough to speak to them, they can shoot arrows well enough to count as expert marksmen, they can move quickly and quietly through the forest like ninjas, etc. That’s just baseline. To have one of the skills listed below means that you have mastered it to a degree to garner a reputation, to earn the respect of your community.

Some skills have sublists. These only become available if you take the core skill first. To gain a subskill means that you have excelled and focused on that specific subskill to the extent that you are renowned as one of the best that the community has ever seen.

* When you master a skill that has one or more subskills, there’s a 50% chance that you learn a subskill next. Otherwise, the subskills are simply added to the list of skills you might pick up next.
* When you master a subskill, there’s a 50% chance that you learn a different subskill listed under the same skill next. Otherwise, the subskills are simply added to the list of skills you might pick up next.

### All skills

* Acting
* Basketweaving
* Building
  * Boats
  * Bridges
  * Houses
* Clothesmaking
* Communication
* Cooking
* Dancing
* Deception
* Deescalation
* History
* Magic
  * Healing
  * Shapeshifting
  * Sleight of hand
  * Sorcery
* Medicine
  * Anatomy
  * Herbal remedies
  * Midwife
  * Surgeon
  * Wounds
* Painting
  * Body art
  * Rock art
  * Tattooing
* Philosophy
  * Epistemology
  * Logic
  * Moral philosophy 
  * Religion
* Playing the drum
* Playing the flute
* Poetry
* Pottery
* Reading
* Running
  * Endurance running
  * Parkour
  * Speed running
* Science
  * Astronomy
  * Botany
  * Chemistry
  * Ecology
  * Geology
  * Physics
  * Meteorology
  * Mycology
  * Zoology
* Scouting
* Singing
* Storytelling
* Toolmaking
  * Arrows
  * Atlatls
  * Axes
  * Blowgun
  * Bolas
  * Boomerangs
  * Bow drills
  * Bowls
  * Bows
  * Fish hooks
  * Hammers
  * Knives
  * Mortars and pestles
  * Needles
  * Ropes
  * Scrapers
  * Slings
  * Spears
  * Spoons
* Weapons
  * Archery
  * Atlatl
  * Axes
  * Blowgun
  * Bolas
  * Boomerangs
  * Slings
  * Spears

### Age restrictions

The following skills are things that you’re less likely to start learning after the age of 56. If you pick one of these, we’ll pick again. But not a second time, so if you pick something like this twice, sure, you can be the guy who starts getting into endurance running at the age of 72. We just don’t want it to be equally likely as a 16 year old.

* Building
* Running
* Scouting
* Weapons

### Magic

Each character has a magical calling, base of zero.

Factor | Effect
---|:---:
Mother knows magic | +10
Father knows magic | +10
Intersex | +10
Third gender | +10
Homosexual | +10
Neurodivergent | +15
Dwarf | +10

If you select magic as the skill you want to learn, check for magical calling. If that check fails, select again. If you select magic a second time, go with it.

Anytime a character receives a sickness or injury random event, also check for magical calling. On a pass, drop whatever skill you were learning and start learning magic instead.

Once you learn magic, its subskills are treated like any other subskills.

### Reading

If you select reading and there’s no one in the community who can read, select again. If you get reading a second time, stick with it.

If you select reading and there is someone else in the community who can read, there’s a 50% chance that she’ll teach you if your intelligence is above average and your extraversion is below average. Otherwise, select again. If you get reading a second time, stick with it.

**Comment by jefgodesky** [comment]
OK, there's quite a bit here, but I think we've laid out enough to get to work. I'm going to split this up into a bunch of new issues, and they're all going to be part of a new milestone: **Community Creation**. That'll help us track how much is left and when it's done.

I'm going to be making a new branch from `develop` called `community-creation`. I'll work on each ticket in its own branch from that using the normal convention (e.g., `features/##-developer-description`), merging them back into `community-creation` when they're finished. When we get to the point of merging `community-creation` back into `develop`, there will be much rejoicing.

**Comment by jefgodesky** [comment]
Thanks for your feedback, everybody. This has been split up into other issues (#37, #38, #39, #40, #41, #42, and #43). This ticket will remain as the "bridge" to mark that we really do need to get community creation done before we can consider the initial launch successful (I don't think issues can belong to two milestones, and there's enough parts to this to warrant its own milestone to keep track of it, so this one is remaining in "Initial Launch" to track that part).
